core/src/main/java/com/amplifyframework/core/model/types/JavaFieldType.java

```diff
@@ -17,6 +17,9 @@
 
 import androidx.annotation.NonNull;
 
+import com.amplifyframework.core.model.AWSDate;
+import com.amplifyframework.core.model.AWSDateTime;
+import com.amplifyframework.core.model.AWSTime;
```

Comment:
yeah `AWSTimestamp` is confusing to me, because the implementation handles it like a `Long` (like it should imo), so I don't fully understand why AWSTimestamp exists...

Reply:
which "implementation" are you referring to?  

The codegen currently generates a model with a `Long`, but we've asked them to change this to `AWSTimestamp`.

Reply:
The SQLite adapter handles as a `Long`.

Maybe there's a reason for using `AWSTimestamp` instead, if that's the preferred way I can change this. But it's worth noting that both iOS and JS handle timestamp as a number. The alternative for a more "controlled type" in those platforms is `AWSDateTime`.